terraform/lib/dependabot/terraform/registry_client.rb

```diff
@@ -114,26 +114,11 @@ def all_module_versions(identifier:)
       sig { params(dependency: Dependabot::Dependency).returns(T.nilable(Dependabot::Source)) }
       def source(dependency:)
         type = T.must(dependency.requirements.first)[:source][:type]
-        base_url = service_url_for(service_key_for(type))
-        case type
-        # https://www.terraform.io/internals/module-registry-protocol#download-source-code-for-a-specific-module-version
-        when "module", "modules", "registry"
-          download_url = URI.join(base_url, "#{dependency.name}/#{dependency.version}/download")
-          response = http_get(download_url)
-          return nil unless response.status == 204
-
-          source_url = response.headers.fetch("X-Terraform-Get")
-          source_url = URI.join(download_url, source_url) if
-            source_url.start_with?("/", "./", "../")
-          source_url = RegistryClient.get_proxied_source(source_url) if source_url
-        when "provider", "providers"
-          response = http_get(URI.join(base_url, "#{dependency.name}/#{dependency.version}"))
-          return nil unless response.status == 200
+        source_url = fetch_source_url(dependency, type)
```

Comment:
Rubocop complained about complexity in this function, so I chose to extract it. If you prefer a different approach, please let me know.